nixos/modules/services/security/cfssl.nix

```diff
@@ -0,0 +1,205 @@
+    };
+
+    intDir = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Intermediates directory";
+    };
+
+    metadata = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = ''
+        Metadata file for root certificate presence.
+        The content of the file is a json dictionary (k,v): each key k is
+        a SHA-1 digest of a root certificate while value v is a list of key
+        store filenames.
+      '';
+    };
+
+    remote = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = "Remote CFSSL server";
+    };
+
+    configFile = mkOption {
+      default = null;
+      type = types.nullOr types.str; # FIXME: types.nonStorePath, may contain secrets
+      description = "Path to configuration file";
+    };
+
+    responder = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Certificate for OCSP responder";
+    };
+
+    responderKey = mkOption {
+      default = null;
+      type = types.nullOr types.str; # FIXME: types.nonStorePath, secret
+      description = "Private key for OCSP responder certificate";
+    };
+
+    tlsKey = mkOption {
+      default = null;
+      type = types.nullOr types.str; # FIXME: types.nonStorePath, secret
+      description = "Other endpoint's CA private key";
+    };
+
+    tlsCert = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Other endpoint's CA to set up TLS protocol";
+    };
+
+    mutualTlsCa = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Mutual TLS - require clients be signed by this CA";
+    };
+
+    mutualTlsCn = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = "Mutual TLS - regex for whitelist of allowed client CNs";
+    };
+
+    tlsRemoteCa = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "CAs to trust for remote TLS requests";
+    };
+
+    mutualTlsClientCert = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Mutual TLS - client certificate to call remote instance requiring client certs";
+    };
+
+    mutualTlsClientKey = mkOption {
+      default = null;
+      type = types.nullOr types.str; # FIXME: types.nonStorePath, secret
+      description = "Mutual TLS - client key to call remote instance requiring client certs";
+    };
+
+    dbConfig = mkOption {
+      default = null;
+      type = types.nullOr types.str; # FIXME: types.nonStorePath, writeable
+      description = "Certificate db configuration file";
+    };
+
+    logLevel = mkOption {
+      default = 1;
+      type = types.enum [ 0 1 2 3 4 5 ];
+      description = "Log level (0 = DEBUG, 5 = FATAL)";
+    };
+  };
+
+  config = {
+    users.extraGroups."cfssl" = {
```

Comment:
Nit: cfssl doesn't need to be quoted, same with the one in `users.extraUsers` below.